### PR TITLE
Update gem to new solidus conventions

### PR DESCRIPTION
**Pull request by kennyadsl**
This PR updates the extension to make it work without deprecated code with latest versions of Solidus.

This extension will not work with Solidus 2.2 anymore but there's solidus_gateway for older versions.

### CODE REVIEW

File: lib/solidus_stripe/engine.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@ class Engine < Rails::Engine
     engine_name 'solidus_stripe'
 
     initializer "spree.stripe.payment_method", after: "spree.register.payment_methods" do |app|
-      app.config.spree.payment_methods << Spree::Gateway::StripeGateway
+      if Gem.loaded_specs.key?('solidus_gateway')
+        app.config.spree.payment_methods.delete(Spree::Gateway::StripeGateway)
+      end
```
Comment (kennyadsl) [inline lib/solidus_stripe/engine.rb]: We can just even check if `Spree::Gateway::StripeGateway` is present into `app.config.spree.payment_methods. ` but I don't know what happens if `solidus_gateway` is loaded after `solidus_stripe`. We'll have no control over removing the old `StripeGateway` from the available payment methods.

**Comment by kennyadsl** [comment]
@tvdeyen what do you think?

File: lib/solidus_stripe/engine.rb (hunk 1 above)
Comment (kennyadsl) [reply lib/solidus_stripe/engine.rb]: No we can't check that or we'll have `undefined constant Spree::Gateway::StripeGateway` 😄

File: db/migrate/20181010123508_update_stripe_payment_method_type_to_credit_card.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class UpdateStripePaymentMethodTypeToCreditCard < SolidusSupport::Migration[5.1]
+  def up
+    Spree::PaymentMethod.where(type: 'Spree::Gateway::StripeGateway').update_all(type: 'Spree::PaymentMethod::StripeCreditCard')
+
+    Spree::Preference.where("#{ActiveRecord::Base.connection.quote_column_name('key')} LIKE 'spree/gateway/stripe_gateway'").each do |pref|
```
Comment (kennyadsl) [inline db/migrate/20181010123508_update_stripe_payment_method_type_to_credit_card.rb]: I have some doubt that this is really useful. 

For example I was using `static_model_preferences` in my app and this is what I had to do to make it work (this code lives into an initializer):

```diff
- Spree::Config.configure do |config|
-   config.static_model_preferences.add(
-     Spree::Gateway::StripeGateway,
-     'stripe_credentials',
-     secret_key: secret_key,
-     publishable_key: publishable_key
-   )
- end

+ Spree::Config.configure do |config|
+   config.static_model_preferences.add(
+     Spree::PaymentMethod::StripeCreditCard,
+     'stripe_credentials',
+     secret_key: secret_key,
+     publishable_key: publishable_key
+   )
+ end
```

**Comment by kennyadsl** [comment]
@tvdeyen do you have any comment here?

**Comment by aitbw** [comment]
Re-running specs, (small) Travis config modifications shouldn't be a cause for a spec to fail

**Pull request merged**